src/main/java/com/puppycrawl/tools/checkstyle/XpathFileGeneratorAstFilter.java

```diff
@@ -0,0 +1,149 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.puppycrawl.tools.checkstyle.api.AuditEvent;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.LocalizedMessage;
+import com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator;
+
+/**
+ * Catches {@code TreeWalkerAuditEvent} and generates corresponding xpath query.
+ * Stores event and query inside static variables for {@code XpathFileGeneratorAuditListener}.
+ * See issue #102 https://github.com/checkstyle/checkstyle/issues/102
+ *
+ */
+public class XpathFileGeneratorAstFilter extends AutomaticBean implements TreeWalkerFilter {
+
+    /** The delimiter between xpath queries. */
+    private static final String DELIMITER = " | \n";
+
+    /** Default distance between tab stops. */
+    private static final int DEFAULT_TAB_WIDTH = 8;
```

Comment:
This check is internally only to us and tab width is always supplied, so I question if this is needed.

Reply:
done